### DISCUSSION TITLE
Looping in the transcribe process

### DISCUSSION THREAD

**Discussion by Sogl** [ggml-org/whisper.cpp#635] [Q&A]
I use large model and Russian language.

Command:
```
./main -m models/ggml-large.bin -f samples/55.wav -l ru -otxt -ovtt -osrt
```

In my example, it happens from 6 to 18 minutes of the lecture and has not yet thought to interrupt:
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/7556290/226551172-6d9d1d50-5572-49ff-9d17-67096df09ca4.png">

I have noticed that looping happens when the lecturer repeats the same phrase (rereads it) several times.

The second person can pull him out of this cycle, in my case it is the one who asks the lecturer a question.

Any idea how to solve this?

**UPDATE.**

The first case lasted 25 minutes and stopped after someone wrote that the lecturer could not be heard.

The second looping started at 35 minutes into the lecture, got to 1 hour and 20 minutes, and I interrupted the process:
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/7556290/226570845-1920097f-6a50-4910-a204-4a4d0d025f18.png">

It is also unclear how to save the already existing progress to files. If the process is interrupted with Ctrl + C, nothing will be written.

**Comment by ulatekh** [comment]
I've been able to reproduce this with a short audio clip (i.e. 3.4 seconds), the medium-en model, and a grammar. I have yet to figure out how to cut down the grammar to a minimum that still reproduces the problem; it freaks out in other ways. (And without the grammar, it's fast and mostly accurate.) But it's very slow when it runs into this problem, as well as highly inaccurate. (The original phrase was "pan 200 miles ENE".)

```
[00:00:00.000 --> 00:00:00.480]   two hundred and two h
[00:00:00.480 --> 00:00:00.960]   two hundred and two h
[00:00:00.960 --> 00:00:01.440]   two hundred and two h
[00:00:01.440 --> 00:00:31.440]   a

whisper_print_timings:     load time =  1485.50 ms
whisper_print_timings:     fallbacks =   4 p /   0 h
whisper_print_timings:      mel time =     6.09 ms
whisper_print_timings:   sample time = 14166.93 ms /   207 runs (   68.44 ms per run)
whisper_print_timings:   encode time =  1958.18 ms /     4 runs (  489.55 ms per run)
whisper_print_timings:   decode time =   263.88 ms /    16 runs (   16.49 ms per run)
whisper_print_timings:   batchd time =  1120.50 ms /   159 runs (    7.05 ms per run)
whisper_print_timings:   prompt time =     0.00 ms /     1 runs (    0.00 ms per run)
whisper_print_timings:    total time = 17559.39 ms

```

Is there anyone that can direct me how to debug this on my side?
